### ISSUE TITLE
@slonik/typegen: feat: extract user defined types

### ISSUE THREAD

**Issue by charlex** [mmkal/slonik-tools#163]
Like this one:

```
    create type my_custom_type as enum (
        'foobar',
        'foobaz',
        'foo',
        'bar'
    );
```

Right now it gets mapped to `unknown`:

```
  my_custom_type: unknown
```

But would be better as:

```
  my_custom_type:
        'foobar' |
        'foobaz' |
        'foo' |
        'bar'
    );
```

Or

```
type MyCustomType = 'foobar' |
        'foobaz' |
        'foo' |
        'bar';

...

my_custom_type: MyCustomType;
```

**Comment by mmkal** [comment]
Support for this is on the todo list. Right now you can workaround by manually specifying a type mapper: 

```typescript
 setupTypeGen({
  ...
  typeMapper: {
    my_custom_type: [`'foobar' | 'foobaz' | 'foo' | 'bar'`, val => val]
  }
})
```
similarly to how `timestamptz` -> `Date` is handled in the demo project: https://github.com/mmkal/slonik-tools/blob/d2ef0fd9da3f23a9e906b0c4ead3bd7af78d92e8/packages/typegen/test/index.test.ts#L175

I'll leave this issue open as a reminder to add some docs around if I can't get a fix out soon.

**Comment by mmkal** [comment]
released in v0.6.0: https://www.npmjs.com/package/@slonik/typegen/v/0.6.0
